api/src/pcapi/core/subscription/api.py

```diff
@@ -547,26 +547,20 @@ def update_user_birth_date(user: users_models.User, birth_date: datetime.date |
         pcapi_repository.repository.save(user)
 
 
-def has_passed_all_checks_to_become_beneficiary(user: users_models.User) -> bool:
-    fraud_check = get_activable_identity_fraud_check(user)
-    if not fraud_check:
-        return False
-
-    if _should_validate_phone(user, fraud_check.eligibilityType):
-        return False
-
-    subscription_item = get_user_profiling_subscription_item(user, fraud_check.eligibilityType)
-    if subscription_item.status in (models.SubscriptionItemStatus.TODO, models.SubscriptionItemStatus.KO):
-        return False
-
-    profile_completion = get_profile_completion_subscription_item(user, fraud_check.eligibilityType)
-    if profile_completion.status != models.SubscriptionItemStatus.OK:
-        return False
-
-    if not fraud_api.has_performed_honor_statement(user, fraud_check.eligibilityType):  # type: ignore [arg-type]
-        return False
+def _get_eligible_and_ok_identity_fraud_check(user: users_models.User) -> fraud_models.BeneficiaryFraudCheck | None:
+    """Finds latest created activable identity fraud check for a user."""
+    user_identity_fraud_checks = [
+        fraud_check
+        for fraud_check in user.beneficiaryFraudChecks
+        if fraud_check.status == fraud_models.FraudCheckStatus.OK
+        and fraud_check.type in fraud_models.IDENTITY_CHECK_TYPES
+        and users_api.is_eligible_for_beneficiary_upgrade(user, fraud_check.eligibilityType)
+        and users_api.is_user_age_compatible_with_eligibility(user.age, fraud_check.eligibilityType)
+    ]
+    if not user_identity_fraud_checks:
+        return None
 
-    return True
+    return sorted(user_identity_fraud_checks, key=lambda fraud_check: fraud_check.dateCreated, reverse=True)[0]
```

Comment:
Est-ce qu'une approche comme

```python
def _get_eligible_and_ok_identity_fraud_check(user: users_models.User) -> fraud_models.BeneficiaryFraudCheck | None:
    """Finds latest created activable identity fraud check for a user."""
    fraud_checks = sorted(
        user.beneficiaryFraudChecks
        key=lambda fraud_check: fraud_check.dateCreated, reverse=True
    )

    for fraud_check in fraud_checks:
        if users_api.fraud_check_is_eligible_and_ok:  # à implémenter (nom à revoir)
            return fraud_check

```

ne serait pas plus lisible ?

Il y a deux points qui me dérangent un peu, mais c'est plus une question de style et de préférence personnelle donc fais-en ce que tu veux :)

1. la _list comprehension_ ( `user_identity_fraud_checks` ) de six lignes n'est pas très lisible ;
2. je trouve ça plus clair dans ce cas de faire une boucle for, plutôt que de construire toute une liste, de la trier et prendre le premier élément tout à la fin.